include/bitcoin/database/impl/record_manager.ipp

```diff
@@ -134,15 +134,20 @@ Link record_manager<Link>::allocate(size_t count)
 template <typename Link>
 memory_ptr record_manager<Link>::get(Link link) const
 {
-    // If record >= count() then we should still be within the file. The
-    // condition implies a block has been unconfirmed while reading it.
+    BITCOIN_ASSERT_MSG(link < count(), "Read past end of file.");
```

Comment:
Yes, the comment above is obsolete (blocks are no longer removed). Otherwise the assertion would be invalid.